homeassistant/components/dremel_3d_printer/binary_sensor.py

```diff
@@ -0,0 +1,83 @@
+"""Support for monitoring Dremel 3D Printer binary sensors."""
+from __future__ import annotations
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Dremel3DPrinterDataUpdateCoordinator, Dremel3DPrinterDeviceEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the available Dremel binary sensors."""
+    coordinator: Dremel3DPrinterDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    device_id = config_entry.unique_id
+
+    assert device_id is not None
+
+    entities: list[BinarySensorEntity] = [
+        Dremel3DPrinterDoorBinarySensor(coordinator, config_entry),
+        Dremel3DPrinterMainBinarySensor(coordinator, config_entry),
+    ]
+
+    async_add_entities(entities)
```

Comment:
This can be rewritten as a generator

Reply:
You can also use `BinarySensorEntityDescription`

Reply:
I looked into Generators but I couldn't figure out how to use them here. Using BinarySensorEntityDescription turns out that adds way more lines of code compared to how it is right now, unless I've done something wrong. Can you provide a quick example on how you'd go with rewriting this?